File: src/main/java/com/app/pages/LoginAppPage.java
package com.app.pages;

import io.appium.java_client.AppiumBy;
import io.appium.java_client.ios.IOSDriver;
import org.openqa.selenium.By;

public class LoginAppPage extends BaseAppPage {

    private By loginInitButton = AppiumBy.name("Login");
    private By emailInput = By.name("Email address");
    private By passwordInput = By.name("Password");
    private By continueButton = By.name("Continue");
    private By interpreterLabel = By.name("  Interpreter");

    public LoginAppPage(IOSDriver driver) {
        super(driver);
    }

    public void clickInitialLogin() {
        click(loginInitButton);
    }

    public void setupContext() {
        switchToContext(0);
    }

    public void enterEmail(String email) {
        writeText(emailInput, email);
    }

    public void enterPassword(String password) {
        writeText(passwordInput, password);
    }

    public void clickContinue() {
        click(continueButton);
    }

    public void selectInterpreter() {
        click(interpreterLabel);
    }

    public void performLogin(String email, String password) {
        clickInitialLogin();
        setupContext();
        enterEmail(email);
        clickContinue();
        enterPassword(password);
        clickContinue();
        selectInterpreter();
    }
}

File: pages/app/LoginAppPage.ts
import { Browser } from "webdriverio";
import BaseAppPage from "./BaseAppPage";

class LoginAppPage extends BaseAppPage {
    private loginInitButton: string;
    private emailInput: string;
    private passwordInput: string;
    private continueButton: string;

    constructor(driver: Browser) {
        super(driver);
        this.loginInitButton = "~Login";
        this.emailInput = "~Email address";
        this.passwordInput = "~Password";
        this.continueButton = "~Continue";
    }

    async clickInitialLogin(): Promise<void> {
        await this.click(this.loginInitButton);
    }

    async setupContext(): Promise<void> {
        await this.switchToContext(0);
    }

    async enterEmail(email: string): Promise<void> {
        await this.writeText(this.emailInput, email);
    }

    async enterPassword(password: string): Promise<void> {
        await this.writeText(this.passwordInput, password);
    }

    async clickContinue(): Promise<void> {
        await this.click(this.continueButton);
    }

    async performLogin(email: string, password: string): Promise<void> {
        await this.clickInitialLogin();
        // await this.setupContext();
        await this.enterEmail(email);
        await this.clickContinue();
        await this.enterPassword(password);
        await this.clickContinue();
    }
}

export default LoginAppPage;
